commit_trees.go

```diff
@@ -192,65 +187,73 @@ func (commitTreesRowKeyMapper) toRow(data []byte) (sql.Row, error) {
 	return sql.Row{repo, commit, tree}, nil
 }
 
-func commitTreesIterBuilder(ctx *sql.Context, selectors selectors, columns []sql.Expression) (RowRepoIter, error) {
-	repos, err := selectors.textValues("repository_id")
-	if err != nil {
-		return nil, err
-	}
-
-	hashes, err := selectors.textValues("commit_hash")
-	if err != nil {
-		return nil, err
-	}
-
-	return &commitTreesIter{
-		ctx:          ctx,
-		commitHashes: hashes,
-		repos:        repos,
-	}, nil
-}
-
-type commitTreesIter struct {
-	ctx  *sql.Context
-	repo *Repository
+type commitTreesRowIter struct {
+	ctx           *sql.Context
+	repo          *Repository
+	skipGitErrors bool
+	index         sql.IndexValueIter
 
 	commits object.CommitIter
 	commit  *object.Commit
 	trees   *object.TreeWalker
 
 	// selectors for faster filtering
-	repos        []string
-	commitHashes []string
+	commitHashes []plumbing.Hash
+	mapper       commitTreesRowKeyMapper
+}
+
+func (i *commitTreesRowIter) Next() (sql.Row, error) {
+	if i.index != nil {
+		return i.nextFromIndex()
+	}
+
+	return i.next()
+}
+
+func (i *commitTreesRowIter) init() error {
+	var err error
+	if len(i.commitHashes) > 0 {
+		i.commits, err = NewCommitsByHashIter(i.repo, i.commitHashes)
+	} else {
+		i.commits, err = i.repo.Repo.CommitObjects()
+	}
+
+	return err
 }
 
-func (i *commitTreesIter) NewIterator(repo *Repository) (RowRepoIter, error) {
-	var commits object.CommitIter
-	if len(i.repos) == 0 || stringContains(i.repos, repo.ID) {
-		var err error
-		commits, err = NewCommitsByHashIter(repo, i.commitHashes)
+var commitTreesHashIdx = CommitTreesSchema.IndexOf("commit_hash", CommitTreesTableName)
```

Comment:
why it's a package `var`? Can be a function?

Reply:
Because this value will be used in other functions